### ISSUE TITLE
Emoji autocompletion should not shadow smileys

### ISSUE THREAD

**Issue by ThiefMaster** [mattermost/mattermost-server#7651]
#### Summary
If you type a "classic" smiley such as `:-D` you get autocompletions for anything matching `:*-d*:`

#### Steps to reproduce
Add a custom emoji named e.g. `:cern-dipole:`, then type `:-D` - if you now press ENTER you get autocompletion for the custom emoji instead of the grinning smiley.

#### Expected behavior
The autocomplete should either not show up at all or include both the smiley and the emoji.

#### Possible fixes
Maybe `-` should require something on both sides so the `:cern-dipole:` example would only be autocompleted if you type `:c-d:`. Also, if there is an exact match with a smiley it might be worth to show that in the autocompletion list too.

**Comment by hmhealey** [comment]
Interesting. We fixed this for shorter smileys such as :p by not triggering the autocomplete until you've typed a colon plus 2 more letters, but that doesn't work in this case. I don't think we want to just increase that number since that'll make it harder to use the autocomplete when you actually want it.

I've filed a Jira ticket ([link](https://mattermost.atlassian.net/browse/PLT-7930)) to look into this. Perhaps there's something we can do like not showing the autocomplete if the text matches a smiley exactly.
